models/bernoulli_connection.h

```diff
@@ -0,0 +1,255 @@
+ * A suitable Connector containing these connections can be obtained from
+ * the template GenericConnector.
+ */
+
+template < typename targetidentifierT >
+class BernoulliConnection : public Connection< targetidentifierT >
+{
+public:
+  // this line determines which common properties to use
+  typedef CommonSynapseProperties CommonPropertiesType;
+
+  typedef Connection< targetidentifierT > ConnectionBase;
+
+  /**
+   * Default Constructor.
+   * Sets default values for all parameters. Needed by GenericConnectorModel.
+   */
+  BernoulliConnection()
+    : ConnectionBase()
+    , weight_( 1.0 )
+    , p_transmit_(1.0)
+  {
+  }
+
+  /**
+   * Copy constructor from a property object.
+   * Needs to be defined properly in order for GenericConnector to work.
+   */
+  BernoulliConnection( const BernoulliConnection& rhs )
+    : ConnectionBase( rhs )
+    , weight_( rhs.weight_ )
+    , p_transmit_( rhs.p_transmit_ )
+  {
+  }
+
+  // Explicitly declare all methods inherited from the dependent base
+  // ConnectionBase. This avoids explicit name prefixes in all places these
+  // functions are used. Since ConnectionBase depends on the template parameter,
+  // they are not automatically found in the base class.
+  using ConnectionBase::get_delay_steps;
+  using ConnectionBase::get_rport;
+  using ConnectionBase::get_target;
+
+
+  class ConnTestDummyNode : public ConnTestDummyNodeBase
+  {
+  public:
+    // Ensure proper overriding of overloaded virtual functions.
+    // Return values from functions are ignored.
+    using ConnTestDummyNodeBase::handles_test_event;
+    port
+    handles_test_event( SpikeEvent&, rport )
+    {
+      return invalid_port_;
+    }
+    port
+    handles_test_event( RateEvent&, rport )
+    {
+      return invalid_port_;
+    }
+    port
+    handles_test_event( DataLoggingRequest&, rport )
+    {
+      return invalid_port_;
+    }
+    port
+    handles_test_event( CurrentEvent&, rport )
+    {
+      return invalid_port_;
+    }
+    port
+    handles_test_event( ConductanceEvent&, rport )
+    {
+      return invalid_port_;
+    }
+    port
+    handles_test_event( DoubleDataEvent&, rport )
+    {
+      return invalid_port_;
+    }
+    port
+    handles_test_event( DSSpikeEvent&, rport )
+    {
+      return invalid_port_;
+    }
+    port
+    handles_test_event( DSCurrentEvent&, rport )
+    {
+      return invalid_port_;
+    }
+  };
+
+  void
+  check_connection( Node& s,
+    Node& t,
+    rport receptor_type,
+    double,
+    const CommonPropertiesType& )
+  {
+    ConnTestDummyNode dummy_target;
+    ConnectionBase::check_connection_( dummy_target, s, t, receptor_type );
+  }
+
+  void
+  send( Event& e, thread t, double, const CommonSynapseProperties& )
+  {
+    SpikeEvent e_spike = static_cast<SpikeEvent &>(e);
+
+    librandom::RngPtr rng = kernel().rng_manager.get_rng( t );
+    ulong n_spikes_in = e_spike.get_multiplicity();
```

Comment:
No, `n_spikes_in` will not be changed in the code below, and
```c++
const size_t n_spikes_in = e_spike.get_multiplicity();
```
will inform the compiler about this and allow further optimization. It also tells a reader of the code that this value will remain const and thus reduces the code complexity for the reader.